src/FakeItEasy/MethodBaseExtensions.cs

```diff
@@ -18,5 +20,119 @@ public static string GetGenericArgumentsCSharp(this MethodBase method)
+        }
+
+        public static string GetDescription(this MethodBase method)
+        {
+            var builder = new StringBuilder();
+
+            builder
+                .Append(method.DeclaringType)
+                .Append(".");
+
+            AppendMethodName(builder, method);
+
+            AppendParameterList(builder, method);
+
+            return builder.ToString();
+        }
+
+        private static void AppendMethodName(StringBuilder builder, MethodBase method)
+        {
+            if (IsPropertyGetterOrSetter(method))
+            {
+                builder.Append(method.Name.Substring(4));
+            }
+            else
+            {
+                builder.Append(method.Name);
+            }
+
+            builder.Append(method.GetGenericArgumentsCSharp());
+        }
+
+        private static void AppendParameterList(StringBuilder builder, MethodBase method)
+        {
+            var parameters = method.GetParameters();
+
+            if (parameters.Length > 0 || !method.IsPropertyGetterOrSetter())
+            {
+                AppendParameterListPrefix(builder, method);
+
+                AppendParameters(builder, parameters);
+
+                AppendParameterListSuffix(builder, method);
+            }
+        }
+
+        private static void AppendParameterListPrefix(StringBuilder builder, MethodBase method)
+        {
+            if (method.IsPropertyGetterOrSetter())
+            {
+                builder.Append("[");
+            }
+            else
+            {
+                builder.Append("(");
+            }
+        }
+
+        private static void AppendParameterListSuffix(StringBuilder builder, MethodBase method)
+        {
+            if (method.IsPropertyGetterOrSetter())
+            {
+                builder.Append("]");
+            }
+            else
+            {
+                builder.Append(")");
+            }
+        }
+
+        private static void AppendParameters(StringBuilder builder, ParameterInfo[] parameters)
+        {
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                AppendParameterSeparator(builder, i, parameters.Length);
+                builder
+                    .Append(parameters[i].ParameterType.FullNameCSharp())
+                    .Append(" ")
+                    .Append(parameters[i].Name);
+            }
+        }
+
+        private static void AppendParameterSeparator(StringBuilder builder, int argumentIndex, int totalNumberOfArguments)
+        {
+            if (totalNumberOfArguments > 2)
+            {
+                if (argumentIndex > 0)
+                {
+                    builder.Append(",");
+                }
+
+                builder.AppendLine();
+                builder.Append("    ");
+                return;
+            }
+
+            if (argumentIndex > 0)
+            {
+                builder.Append(", ");
+            }
+        }

```

Comment:
We probably don't want to split on multiple lines in this scenario. Should I add a parameter to control this? Or just assume we never want to split on multiple lines?